src/com/mitrais/dal/AccountDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package src.com.mitrais.dal;
+
+import src.com.mitrais.model.Account;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AccountDaoImpl implements AccountDao {
+    List<Account> accountList = new ArrayList<Account>();
+    private static AccountDaoImpl accountDaoInstance;
+
+    public static AccountDaoImpl getInstance(){
+        if(accountDaoInstance == null){
+            accountDaoInstance = new AccountDaoImpl();
+        }
+        return  accountDaoInstance;
+    }
+
+    public AccountDaoImpl(){
+        accountList.add(new Account("John Doe", "012108", 100, "112233"));
+        accountList.add(new Account("Jane Doe", "932012", 30, "112244"));
+    }
+
+    @Override
+    public List<Account> getAccounts() {
+        return accountList;
+    }
+
+    @Override
+    public Account getLoginAccount(String accNumber, String pin) {
+        return accountList.stream().filter(account -> accNumber.equals(account.getAccountNumber()) && pin.equals(account.getPin())).findAny().orElse(null);
```

Comment:
If you return an Optional<Account> and throw an exception here, you don't really need to validate for nulls outside. Take it into consideration to improve your error handling

Reply:
error handling improved.

Reply:
Instead of throwing a generic Runtime, throw your own